File: VB/Models/Invoice.vb
Imports System
Imports System.Collections.Generic

Public Class Invoice
	Private Const DataItemsCount As Integer = 20

	Public Property Id() As Integer
	Public Property Description() As String
	Public Property Price() As Decimal
	Public Property RegisterDate() As Date

	Public Shared Function GetData() As List(Of Invoice)
		Dim invoices As New List(Of Invoice)()

		For i As Integer = 0 To DataItemsCount - 1
			invoices.Add(New Invoice() With {.Id = i, .Description = "Invoice" & i.ToString(), .Price = i * 10, .RegisterDate = Date.Today.AddDays(i - DataItemsCount)})
		Next i

		Return invoices
	End Function
End Class

File: CS/Models/Invoice.cs
using System;
using System.Collections.Generic;

public class Invoice {
    const int DataItemsCount = 20;
    
    public int Id { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public DateTime RegisterDate { get; set; }

    public static List<Invoice> GetData() {
        List<Invoice> invoices = new List<Invoice>();

        for (int i = 0; i < DataItemsCount; i++) {
            invoices.Add(new Invoice() {
                Id = i,
                Description = "Invoice" + i.ToString(),
                Price = i * 10,
                RegisterDate = DateTime.Today.AddDays(i - DataItemsCount)
            });
        }

        return invoices;
    }
}